### PR TITLE
changed color of github icon and refactored code  #3034

### PR DESCRIPTION
**Pull request by dewanhimanshu**
Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x] code is in uniquely-named feature branch and has no merge conflicts
* [x] PR is descriptively titled
* [x] PR body includes `fixes #0000`-style reference to original issue #
* [x] ask `@publiclab/reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software

We have a loose schedule of reviewing and pulling in changes every Tuesday and Friday, and publishing changes on Fridays. 

Thanks!

### CODE REVIEW

**Comment by dewanhimanshu** [comment]
screenshot from 2018-07-11 19-45-19
screenshot from 2018-07-11 19-43-11
screenshot from 2018-07-11 19-40-59
screenshot from 2018-07-11 19-40-50
screenshot from 2018-07-11 19-33-20

**Comment by dewanhimanshu** [comment]
i have made partial and also changed color of github icons to black

**Comment by jywarren** [comment]
This looks great! Did you want to try tweaking the Twitter color to be more
Twitter-y? A little more turquoise, right? I bet the official colors are
posted somewhere.

I really appreciate the attention to detail!!!

**Comment by dewanhimanshu** [comment]
ok @jywarren  i will tweak that

**Comment by dewanhimanshu** [comment]
how is this? I took official colors hex codes from internet
screenshot from 2018-07-11 21-13-21

**Comment by jywarren** [comment]
hmm, looks good... what do other folks think? Also, is it possible to
ensure they're all the same width?

**Comment by SidharthBansal** [comment]
@dewanhimanshu can you please 
1) shift the provider up on all the 4 pages --> Login page in one of the prs was made above and header too.
2) solve the conflicts which may arise with this pr and #3044 in this pr? 
3) make the cell padding between the providers same
They are looking awesome btw

**Comment by dewanhimanshu** [comment]
@SidharthBansal thanks will do the above changes.

**Comment by SidharthBansal** [comment]
In #3044 I just set the icons inside the form. Please post the screenshots once all the changes are made. Also post screenshot for #3044.

**Comment by dewanhimanshu** [comment]
screenshot from 2018-07-11 21-59-31
screenshot from 2018-07-11 21-48-23
screenshot from 2018-07-11 21-46-17
screenshot from 2018-07-11 21-46-09
screenshot from 2018-07-11 21-43-44

**Comment by dewanhimanshu** [comment]
@SidharthBansal  i am  made my branch from latest master branch . #3044 is not merged till now i think so

**Comment by dewanhimanshu** [comment]
screenshot from 2018-07-11 22-19-26
center allign the text

**Comment by SidharthBansal** [comment]
Changes looks good.
#3044 I will do the changes. I will ping you when needed there.
Can you please add or in between on header?
Can you please center align the buttons on login page they are a little towards left

**Comment by dewanhimanshu** [comment]
screenshot from 2018-07-11 22-28-49
screenshot from 2018-07-11 22-24-10

**Comment by dewanhimanshu** [comment]
@SidharthBansal  #3044 after this is merged  i will need your help as i dont know how to pull lastest changes (i am new to github). I red about how to rebase but messed in that

**Comment by SidharthBansal** [comment]
Please ignore #3044 now. I will fix that one.
Yes, you can contact me when this PR is merged and you will be free. I will help you for sure

**Comment by jywarren** [comment]
Yeah this is really really nice. Good teamwork, too!

One small last change would be that in the dropdown, a little extra space
between the buttons and the OR would be nice, just so it doesn't seem
cramped. Then this is ready for merge!

**Comment by dewanhimanshu** [comment]
screenshot from 2018-07-12 00-23-09

**Comment by dewanhimanshu** [comment]
@SidharthBansal  Thank you for guiding me

**Comment by SidharthBansal** [comment]
@jywarren I guess we are now done with this.

**Comment by SidharthBansal** [comment]
We are not done with this. Wait.

**Comment by SidharthBansal** [comment]
Can you please do ``git pull --rebase``?

**Comment by SidharthBansal** [comment]
You actually need to shift the changes from app/view/users/new.html.erb to the _form.html.erb and see the issue linked to #3044

**Comment by jywarren** [comment]
I'm also worried to overlap with https://github.com/publiclab/plots2/pull/3021 - I want to merge that in as we really appreciate the work done there too. But it will create a merge conflict with this PR. @dewanhimanshu would you be OK if I merged https://github.com/publiclab/plots2/pull/3021 and then you rebased over it and made sure it used the partial you created? It's important to us that every contribution be valued, even if they end up overlapping, and I want to be sure the contributor over there doesn't feel left out. I'm sure you understand!

**Comment by dewanhimanshu** [comment]
@jywarren  no problem you can merge #3021 . I will make changes

**Comment by jywarren** [comment]
OK, thank you so much!!! 🙌

**Comment by SidharthBansal** [comment]
Now we need to make sure that #3044 and #3021 changes are there in this pr

**Comment by dewanhimanshu** [comment]
@SidharthBansal  i have resolved the conflicts

**Comment by SidharthBansal** [comment]
@dewanhimanshu can you please try to produce error on signup page and post the screenshots and it is well written on the page like I did in #3044

**Comment by SidharthBansal** [comment]
Please see the issue of #3044. It must be resolved

**Comment by dewanhimanshu** [comment]
screenshot from 2018-07-12 12-52-00

File: app/views/layouts/_header.html.erb [hunk 1]
```diff
@@ -151,16 +151,31 @@
...
+                <div class="form-group">
+                 <% if Rails.env == "production" %>
+                   <label style="display: block; text-align: center;">Log In with</label>
+                    <div class="loginbutton" style="float:left">
+                       <%= render :partial => "layouts/social_icons" %>
+                    </div>
+                    <br style="clear:both;"/>
+                   <label style="display: block; text-align: center; margin-top:13px;">OR</label>
+                  <% end %>
+
```
Comment (SidharthBansal) [inline app/views/layouts/_header.html.erb]: Please remove empty lines

**Comment by dewanhimanshu** [comment]
@SidharthBansal  Thank you

**Comment by jywarren** [comment]
Fantastic!!! 👍 🙌 ⚡️

**Comment by dewanhimanshu** [comment]
thank you

**Comment by dewanhimanshu** [comment]
Looking forward to solve more issues

**Comment by jywarren** [comment]
We'd love that! We're especially looking to close out as many recent bugs as possible:  https://github.com/publiclab/plots2/labels/bug

Thanks!!

**Pull request merged**